agent/05_agent_configure.sh

```diff
@@ -245,7 +245,122 @@ function generate_cluster_manifests() {
   ansible-playbook -vvv \
           -e install_path=${SCRIPTDIR}/${INSTALL_CONFIG_PATH} \
           "${SCRIPTDIR}/agent/create-manifests-playbook.yaml"
+}
+
+function add_haproxy_server_lines() {
+  num_servers=${1}
+  type=${2}
+  port=${3}
+
+  # AGENT_NODES_IPS has master ip addresses listed first
+  # and worker ip addresses listed second.
+  # Depending on the $type, here we find the right
+  # slice of the array to iterate over.
+  if [[ "$type" == "master" ]]; then
+    starting=0
+  else
+    # $type == "worker"
+    starting=$NUM_MASTERS
+    num_servers=$((NUM_MASTERS + num_servers))
+  fi
+
+  for (( n=$starting; n<${num_servers}; n++ ))
+  do
+    cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+  server ${type}-$n ${AGENT_NODES_IPS[n]}:${port} check inter 1s
+EOF
+  done
+}
+
+function enable_load_balancer() {
+  api_ip=${1}
+  load_balancer_ip=${2}
+
+  if [[ "${AGENT_PLATFORM_TYPE}" == "none" && "${NUM_MASTERS}" > "1" ]]; then
+
+    # setup haproxy as the load balancer
+    if [[ "${IP_STACK}" == "v6" ]]; then
+      # The "wildcard" is different depending on IP stack.
+      # See http://docs.haproxy.org/1.6/configuration.html#4.2-bind
+      export HAPROXY_WILDCARD="[::]"
+    else
+      export HAPROXY_WILDCARD="*"
+    fi
+
+    cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+defaults
+    mode                    tcp
+    log                     global
+    timeout connect         10s
+    timeout client          1m
+    timeout server          1m
+frontend stats
+  bind *:1936
+  mode            http
+  log             global
+  maxconn 10
+  stats enable
+  stats hide-version
+  stats refresh 30s
+  stats show-node
+  stats show-desc Stats for ocp4 cluster
+  stats auth admin:ocp4
+  stats uri /stats
+listen api-server-6443
+  bind ${HAPROXY_WILDCARD}:6443
+  mode tcp
+EOF
+    add_haproxy_server_lines $NUM_MASTERS "master" "6443"
 
+    cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+listen machine-config-server-22623
```

Comment:
```suggestion
listen machine-config-server-${AGENT_MCS_PORT}
```

(and also the other occurrences(

Reply:
The port numbers aren't meant to be changed. Let me remove the environment variables and just list the port numbers in 02_configure_host.sh.